scopes/react/react/jest/transformer.js

```diff
@@ -1,8 +1,26 @@
 /* eslint-disable global-require */
 const { transform } = require('@babel/core');
 
+const hasJsxRuntime = (() => {
+  if (process.env.DISABLE_NEW_JSX_TRANSFORM === 'true') {
+    return false;
+  }
+
+  try {
+    require.resolve('react/jsx-runtime');
+    return true;
+  } catch (e) {
+    return false;
+  }
+})();
+
 const presets = [
-  require('@babel/preset-react'),
+  [
+    require('@babel/preset-react'),
+    {
+      runtime: hasJsxRuntime ? 'automatic' : 'classic',
```

Comment:
as continue to previous comments, since it will always be resolved, you can just use automatic here